### PR TITLE
A variety of vehicle tweaks (And some minor CnH changes)

### PR DESCRIPTION
**Pull request by X0-11**
:cl: XO-11
tweak: Vehicle weapon burst sizes have been modified to take advantage of click-drag aim.
tweak: Lekgolo are practically vehicles so: Slight changes to turn speed and weapon fire rate.
tweak: Lekgolo are now immune to the confusion effect of explosions.
rscadd: Most vehicles can now deploy smoke countermeasures to obscure an attack or retreat. Smoke deploys to the front of the vehicle.
tweak: Moves around some breaching charge counts so normal marines get access to them.
/:cl:

### CODE REVIEW

File: code/modules/halo/vehicles/vehicle_action_buttons.dm [hunk 1]
```diff
@@ -0,0 +1,58 @@
+
+#define ACTION_USE_BRAKE "Toggle Brake"
```
Comment (bloxgate) [inline code/modules/halo/vehicles/vehicle_action_buttons.dm:2]: There should be some matching `#undef`s somewhere

File: code/modules/halo/machinery/gunvendors.dm [hunk 2]
```diff
@@ -36,14 +36,20 @@
...
+					/obj/item/weapon/plastique/breaching/longrange = 0,
 					"Miscellaneous" = -1,
 					/obj/item/weapon/armor_patch = 0,
 					/obj/item/weapon/armor_patch/mini = 0,
 					/obj/item/drop_pod_beacon = 0
 					)
 	amounts = list(\
 		/obj/item/weapon/grenade/frag/m9_hedp = 15,
-		/obj/item/weapon/grenade/smokebomb = 15
+		/obj/item/weapon/grenade/smokebomb = 15,
```
Comment (bloxgate) [inline code/modules/halo/machinery/gunvendors.dm:49]: Might be worth at least mentioning there were some vendor amount tweaks in CL?

**Review by bloxgate — CHANGES_REQUESTED** [review]

File: code/modules/halo/machinery/gunvendors.dm (hunk 2 above)
Comment (X0-11) [reply code/modules/halo/machinery/gunvendors.dm:49]: Will do.

File: code/modules/halo/vehicles/vehicle_action_buttons.dm (hunk 1 above)
Comment (X0-11) [reply code/modules/halo/vehicles/vehicle_action_buttons.dm:2]: Added.

**Review by bloxgate — APPROVED** [review]

**Pull request merged**